lib/sqlalchemy/orm/events.py

```diff
@@ -1730,7 +1864,12 @@ def do_something(session, previous_transaction):
 
         """
 
-    def before_flush(self, session, flush_context, instances):
+    def before_flush(
+        self,
+        session: Session,
+        flush_context: UOWTransaction,
+        instances: Optional[Sequence[Any]],
```

Comment:
**mike bayer** (zzzeek) wrote:

instances is probably `Optional[Sequence[_O]]`

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4326